### ISSUE TITLE
S4829 FP: Using Console.In inside a nameof

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6282]
### Description

False Positive for `ReadingStandardInput` (S4829), when using `Console.In` inside a `nameof`.

### Repro steps

```csharp
var consoleIn = nameof(Console.In); // Noncompliant
```

### Expected behavior

It should not be raising an issue, since it is just a compile-time evaluation to a string ("In")

### Actual behavior

It raises an issue.

**Comment by pavel-mikula-sonarsource** [comment]
Rule has been deperecated and deleted in #8049
